packages/shell-evaluator/src/shell-evaluator.ts

```diff
@@ -123,7 +123,7 @@ class ShellEvaluator {
         this.saveState();
         const rewrittenInput = this.asyncWriter.compile(input);
         this.bus.emit(
-          'mongosh:rewrittenAsyncInput',
```

Comment:
We could emit a little more intelligently here if you want - we could only emit when the rewritten input does not match the original input. (We'd have to set a flag to babel to not 'prettify' the generated code, or have the async-rewriter indicate that it  has inserted an await, or remove all whitespace before comparison).

As it is, it will emit an event for literally everything that gets run in the shell, so it might not be that useful.

Reply:
^ that would be heaps more useful to us and to other users!